### DISCUSSION TITLE
ISR Content Reverting on Cloud Run & Scaling

### DISCUSSION THREAD

**Discussion by JordanTheriault** [vercel/next.js#33541] [Help]
I have a website that is using the SSG and ISR capabilities of the framework. It's deployed to Cloud Run on GCP with the following settings:

```
--platform: managed
--cpu: 2
--memory: 8Gi 
--ingress: internal-and-cloud-load-balancing
--port: 3000
--min-instances: 1
--max-instances: 1
--region: northamerica-northeast1
```

It's connected to Contentful for the headless CMS, and set to revalidate every 4 minutes. However, I'm finding that a day or two later my changes that were updated via ISR will revert to the version of the page on original build time (not even the previous version, but the original version). Any ideas on how to fix this issue? Is it because the Cloud Run CPU idles?

Also, I'm trying to scale but replicating a pod will mean the replication will have the same content as the original build. How do I specify a shared volume for the cache?

Thank you!

**Comment by djohalo2** [comment]
@JordanTheriault Did you find a solution for this problem?

I have the same issue and I'm pretty sure it's because the Docker image that is deployed contains a generated cache on build time. Every time a Cloud Run container is activated it serves the files from the Docker image and therefore ignores any file that any other container generated and cached. This kind of defeats the purpose of ISR as containers are scaled up and down all the time... I do use a CDN that fixes and the performance is still great, but it's a shame that a container setup like Cloud Run can't fully use the power and efficiency of Next ISR.

**Reply by gboston** [reply]
@djohalo2 how does the CDN ensure that no outdated (/ build-time) content is shown? I don't think you can solve this easily without having a shared cache (eg. Redis) for these generated files. Unfortunately only a filesystem is supported as a caching mechanism for Nextjs ISR, which means is not compatible for Cloud Run

**Reply by djohalo2** [reply]
@gboston Something went wrong in that sentence. I meant that I configured a Cloud CDN to generally improve caching of statics and get some load of the Cloud Run instances. But you are correct it does not solve the problem of outdated content.

I did find https://cloud.google.com/run/docs/tutorials/network-filesystems-fuse that talks about mounting a Google Cloud Storage network filesystem with Cloud Run. This might be a possible solution. What do you think?

**Reply by gboston** [reply]
I've also looked into this. But it just looks as it is not compatible with a Cloud Native approach if you want to make use of ISR. Build-time files and cache (no idea where the ISR files go at runtime?) should be separate so you could easily clear the cache if a new build is pushed.

I've experimented with a Compute Engine with a Docker container and a volume mounted for the .next directory, but you still need to push the build time files to this volume to ensure you have the latest build. So you need to 2 processes: one to push the static build files and one for your container.

**Reply by macrozone** [reply]
@gboston did you have some progress on this?

do you know whether its possible to specify the diretory for isr?

**Comment by szsebasa1** [comment]
Could any of you solve this problem? Could be easily solved with a 'blocking' fallback for pages that don't have dynamic routes, but I haven't found a way to do that
